jwt/jwt.go

```diff
@@ -147,12 +146,12 @@ func getJWKSWithCache(ctx context.Context) (*clerk.JSONWebKeySet, error) {
 }
 
 // Fetches the JSON web key set from the API and caches it.
-func forceGetJWKS(ctx context.Context, cacheKey string) (*clerk.JSONWebKeySet, error) {
-	jwks, err := jwks.Get(ctx, &jwks.GetParams{})
+func forceGetJWKS(ctx context.Context, client *jwks.Client) (*clerk.JSONWebKeySet, error) {
+	jwks, err := client.Get(ctx, &jwks.GetParams{})
```

Comment:
💡 Not sure, but I would expect that if a Client has caching enabled, then calls to `Get` can either fetch a cached entry or perform an HTTP request and add the cache entry. So in a sense this concern could be transferred to the client itself. I believe that could make the Client more directly usable for other usages, but I may be missing something.

Reply:
You're probably right. I also thought about this, but then I thought, what if we add more methods to the jwks.Client? Would we expect every fetching method to cache results? Would it make sense to name the cache `GetCache` to make it more obvious?

It feels weird to attach a cache and leave it to the responsibility of the consumer how it will be used.

On the other hand, caching is used only for `jwt.Verify`, so it makes sense that this method should be responsible to use the cache as it sees fit. Making decisions about when to cache, what to cache, how long until the cache expires, etc...

Also, it might make more sense to keep the Client.Get operation free of any caching. You can use the cache when you want, but you can also force-fetch.

I was conflicted to be honest. Perhaps your comment enforces the idea that "a jwks.Client with a cache will only fetch after 1h" and that's it?

Reply:
> Would it make sense to name the cache GetCache to make it more obvious?

I think renaming the setting to `EnableReadCache` or `EnableGetCache` if you want to be more exact, sounds fine to me. If I'm thinking this right, you then might be able to hide the `Cache` interface entirely, so it becomes completely internal.

> It feels weird to attach a cache and leave it to the responsibility of the consumer how it will be used.

That's my intuition too. It feels that the storage layer is part of the client but the caller must perform all interactions with the caching layer.

> You can use the cache when you want, but you can also force-fetch.

Wouldn't you just create a new Client instance then? What's the scenario where you need to force fetch considering that you've made the decision to enable caching? Either way, you can maybe expose a `ResetCache` method.